### PR TITLE
Dpl 538 traction select component

### PR DESCRIPTION
**Pull request by seenanair**
- Implementation of traction-select component using tailwind styling
- Removed bootstrap select component
- Unit test case for traction-select
- Updated all references for traction-select
- Updated all related test cases

### CODE REVIEW

File: src/components/labelPrinting/LabelPrintingForm.vue [hunk 1]
```diff
@@ -168,6 +159,7 @@ export default {
       return listSuffixedBarcodes
     },
     suffix() {
+      console.log('SELECTED-SUFFIX=' + this.form.selectedSuffix)
```
Comment (BenTopping) [inline src/components/labelPrinting/LabelPrintingForm.vue:162]: remove console.log

File: src/components/pacbio/PacbioRunWellEdit.vue [hunk 2]
```diff
@@ -41,7 +41,7 @@
           v-model="currentWell.generate_hifi"
           data-attribute="generate-hifi"
           :options="generateHifiOptions[currentRun.system_name]"
-          @change="updateCCSAnalysisOutput"
+          @chaiinputnputnge="updateCCSAnalysisOutput"
```
Comment (BenTopping) [inline src/components/pacbio/PacbioRunWellEdit.vue:44]: not sure what this is?

Comment (BenTopping) [reply src/components/pacbio/PacbioRunWellEdit.vue:44]: I think this is just suppose to be @input

File: src/components/shared/LibraryTypeSelect.vue [hunk 3]
```diff
@@ -71,7 +71,7 @@ export default {
         return []
       }
     },
-    nullOption: ({ allowNone }) => (allowNone ? [{ value: null, text: 'None' }] : []),
+    nullOption: ({ allowNone }) => (allowNone ? [{ value: '', text: 'None' }] : []),
```
Comment (BenTopping) [inline src/components/shared/LibraryTypeSelect.vue:74]: maybe rename to emptyOption as this no longer returns null

File: src/components/shared/LibraryTypeSelect.vue [hunk 4]
```diff
@@ -20,8 +20,8 @@ import { filterByAttribute, mapAttribute } from '@/api/JsonApi'
 // select elements can't handle the former. So we encode it
 const UNDEFINED = '_undefined'
 
-const encode = (value) => (value === undefined ? UNDEFINED : value)
-const decode = (value) => (value === UNDEFINED ? undefined : value)
+const encode = (value) => (value === undefined ? 'UNDEFINED' : value)
```
Comment (BenTopping) [inline src/components/shared/LibraryTypeSelect.vue:23]: Why is undefined in quotes here?

File: src/components/shared/TractionSelect.vue [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  <div class="flex flex-col">
+    <label v-if="title">{{ title }}</label>
+    <select
+      v-if="options"
+      v-bind="$attrs"
+      :value="value"
+      :class="`w-full py-2 px-3 border border-gray-300 bg-white rounded-md shadow-sm focus:outline-none focus:ring-sdb-100 focus:border-sdb-100 disabled:opacity-75 disabled:bg-gray-200 disabled:cursor-not-allowed ${classes}`"
+      @change="input($event)"
+    >
+      <option v-if="optionHeader" value="" disabled="true">{{ optionHeader }}</option>
```
Comment (BenTopping) [inline src/components/shared/TractionSelect.vue:23]: This header is a nice refactor!

File: tests/unit/components/saphyr/SaphyrEnzymeModal.spec.js [hunk 6]
```diff
@@ -39,7 +39,7 @@ describe('SaphyrEnzymeModal.vue', () => {
     it('has enzyme options', async () => {
       let enzymeOptions = {
         enzymeOptions: [
-          { value: null, text: 'Please select an option' },
+          { value: '', text: 'Please select an option' },
```
Comment (BenTopping) [inline tests/unit/components/saphyr/SaphyrEnzymeModal.spec.js:42]: This variable is a bit weird. Maybe remove the enzymeOptions object and keep the array. Then change lines 48/49 to be `wrapper.setData({enzymeOptions: enzymeOptions})
      expect(enzymeModal.enzymeOptions).toEqual(enzymeOptions)`

File: src/components/labelPrinting/LabelPrintingForm.vue [hunk 7]
```diff
@@ -61,8 +59,6 @@
               id="printer-choice"
               v-model="form.selectedPrinterName"
               :options="printerOptions"
-              value-field="text"
-              required
```
Comment (BenTopping) [inline src/components/labelPrinting/LabelPrintingForm.vue:65]: is this no longer `required`?

**Review by BenTopping — CHANGES_REQUESTED** [review]

File: src/components/pacbio/PacbioRunWellEdit.vue (hunk 2 above)
Comment (seenanair) [reply src/components/pacbio/PacbioRunWellEdit.vue:44]: yes 👍

File: src/components/shared/LibraryTypeSelect.vue (hunk 3 above)
Comment (seenanair) [reply src/components/shared/LibraryTypeSelect.vue:74]: 👍

File: src/components/shared/LibraryTypeSelect.vue (hunk 4 above)
Comment (seenanair) [reply src/components/shared/LibraryTypeSelect.vue:23]: Good finding. It's a typo, now changed 👍

File: tests/unit/components/shared/TractionSelect.spec.js [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+  it('sets selected value for select', async () => {
+    const wrapper = buildWrapper({ options: ['Option 1', 'Option 2', 'Option 3'] })
+    const options = wrapper.find('select').findAll('option')
+    await options.at(1).setSelected()
+    expect(wrapper.find('option:checked').element.value).toBe('Option 2')
+    expect(wrapper.find('select').element.value).toEqual('Option 2')
+  })
+
+  it('updates the v-model value when changing the selected value, ', async () => {
+    var wrapper = mount({
```
Comment (stevieing) [inline tests/unit/components/shared/TractionSelect.spec.js:81]: Any reason why you are using var here? Could it just be const?

File: src/components/pacbio/PacbioRunWellDefaultEdit.vue [hunk 9]
```diff
@@ -193,32 +188,12 @@ export default {
...
-          'Do Not Generate',
-        ],
-        'Sequel IIe': [
-          { text: 'Please select a value', value: '', disabled: true },
-          'In SMRT Link',
-          'Do Not Generate',
-          'On Instrument',
-        ],
+        '': [],
+        'Sequel I': ['In SMRT Link', 'Do Not Generate'],
```
Comment (stevieing) [inline src/components/pacbio/PacbioRunWellDefaultEdit.vue:194]: Nice clean up. I will check but have a feeling we can clean up further by remove Sequel I.

File: src/components/shared/index.js [hunk 10]
```diff
@@ -116,10 +115,10 @@ const registerGlobal = (vue) => {
   vue.component('TractionMutedText', TractionMutedText)
   vue.component('TractionToggle', TractionToggle)
   vue.component('TractionSpinner', TractionSpinner)
-  // This is currently causing a bug reverted to Bootstrap component.
   vue.component('TractionInput', TractionInput)
   vue.component('TractionFieldError', TractionFieldError)
   vue.component('TractionResultIcon', TractionResultIcon)
+  vue.component('TractionSelect', TractionSelect)
```
Comment (stevieing) [inline src/components/shared/index.js:121]: This is where you could add the use of the feature flag.

**Review by stevieing — CHANGES_REQUESTED** [review]
Looking good. Just a couple of minor comments.

Lot's of comments and specific tests.

File: tests/unit/components/shared/TractionSelect.spec.js (hunk 8 above)
Comment (seenanair) [reply tests/unit/components/shared/TractionSelect.spec.js:81]: No, changed to const. 👍

File: src/components/pacbio/PacbioRunWellDefaultEdit.vue (hunk 9 above)
Comment (seenanair) [reply src/components/pacbio/PacbioRunWellDefaultEdit.vue:194]: Do you mean combining 'Sequel I' and 'Sequel II' as they both needs same options?

**Review by stevieing — DISMISSED** [review]
All good. I have tested locall and option groups are working.

**Comment by seenanair** [comment]
> All good. I have tested locall and option groups are working.

Thank you

**Review by BenTopping — APPROVED** [review]

**Review by stevieing — APPROVED** [review]

**Pull request merged**